### PR TITLE
fix(eventbus): NATS eventbus auto reconnect and code refactory

### PR DESCRIPTION
**Pull request by whynowy**
1. Fixed a bug that `()` in sensor dependency expression was trimmed when converting group expr to dependency expr.
2. 2 layers of event bus auto reconnect: 
  a. utilize NATS reconnect feature - for short term connection lost such as NATS POD crash
  b. Extra Goroutins in eventsource and sensor service to auto reconnect - for long term connection lost
3. Code refactory to use `ctx.Done` instead of `stopCh`

### CODE REVIEW

File: eventbus/driver/nats.go [hunk 1]
```diff
@@ -73,7 +71,16 @@ func NewNATSStreaming(url, clusterID, subject, clientID string, auth *Auth, logg
 
 func (n *natsStreaming) Connect() (Connection, error) {
 	log := n.logger.WithField("clientID", n.clientID)
-	opts := []nats.Option{}
+	opts := []nats.Option{
+		nats.MaxReconnects(-1),
```
Comment (VaibhavPage) [inline eventbus/driver/nats.go:75]: what's the significance of `-1`?

File: eventsources/eventing.go [hunk 2]
```diff
@@ -261,7 +286,7 @@ func (e *EventSourceAdaptor) Start(ctx context.Context, stopCh <-chan struct{})
 				return err
 			}
 			go func(s EventingServer) {
-				err := s.StartListening(cctx, stopCh, func(data []byte) error {
+				err := s.StartListening(cctx, func(data []byte) error {
```
Comment (VaibhavPage) [inline eventsources/eventing.go:289]: You can now remove // TODO: Need Retry

Comment (whynowy) [reply eventsources/eventing.go:289]: done.

File: eventbus/driver/nats.go (hunk 1 above)
Comment (whynowy) [reply eventbus/driver/nats.go:75]: `-1` means retry forever.

**Pull request merged**